backend/api/student/simulation_instances.py

```
@@ -3,41 +3,281 @@
+    - instance_id (optional)
+    
+    Returns:
+        SQLAlchemy Query object that can be further filtered before calling first() or all()
+    """
+    query = db.query(StudentSimulationInstance).join(
+        UserProgress, StudentSimulationInstance.user_progress_id == UserProgress.id
+    ).join(
+        Scenario, UserProgress.scenario_id == Scenario.id
+    ).filter(
+        StudentSimulationInstance.student_id == student_id,
+        Scenario.is_draft == False,  # Only published simulations
```

Ruff (0.13.3)

Avoid equality comparisons to `False`; use `not Scenario.is_draft:` for false checks

Replace with `not Scenario.is_draft`

(E712)